From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Sun, 27 Jun 2021 20:30:37 +0200
Subject: [PATCH] cloudplane configuration


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index b8c380dae58d47514d684a6de2fd17b7d5135ab1..ec8721fa6536b0297c1ecef7701994474cc9433c 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1694,6 +1694,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
             throw new UnsupportedOperationException("Not supported yet.");
         }
 
+        // CloudPlane start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getCloudPlaneConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // CloudPlane end
+
         @NotNull
         public java.util.Properties getServerProperties() {
             throw new UnsupportedOperationException("Not supported yet.");
