From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <github@booky.tk>
Date: Fri, 2 Jul 2021 13:49:02 +0200
Subject: [PATCH] api methods for setting explosion resistance


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 34902835cd4e030902feb568a2315c7f5d60f0eb..91c8389b456b380c168b17cf01940d058d8a3f6b 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2081,4 +2081,14 @@ public final class Bukkit {
         return server.isLagging();
     }
     // Purpur end
+
+    // CloudPlane start
+    public static float getExplosionResistance(Material material) {
+        return server.getExplosionResistance(material);
+    }
+
+    public static void setExplosionResistance(Material material, float explosionResistance) {
+        server.setExplosionResistance(material, explosionResistance);
+    }
+    // CloudPlane end
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index ec8721fa6536b0297c1ecef7701994474cc9433c..6285466806ea6d6f734335c9445cfd0318dfc4fd 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1833,4 +1833,10 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
      */
     boolean isLagging();
     // Purpur end
+
+    // CloudPlane start
+    float getExplosionResistance(Material material);
+
+    void setExplosionResistance(Material material, float explosionResistance);
+    // CloudPlane end
 }
