From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Fri, 25 Jun 2021 22:20:11 +0200
Subject: [PATCH] added fly command


diff --git a/src/main/java/org/bukkit/util/permissions/CloudPlanePermissions.java b/src/main/java/org/bukkit/util/permissions/CloudPlanePermissions.java
index b84ac9da6cd911c791bd624a17d5edbfdfbc2e95..23cf9352dacfe5d7dd367aba2432f290a5f0f54a 100644
--- a/src/main/java/org/bukkit/util/permissions/CloudPlanePermissions.java
+++ b/src/main/java/org/bukkit/util/permissions/CloudPlanePermissions.java
@@ -13,6 +13,8 @@ public class CloudPlanePermissions {
         Permission cloudplane = DefaultPermissions.registerPermission(ROOT, "Gives the user the ability to use alle CloudPlane utilities and commands", PermissionDefault.FALSE);
 
         Permission commands = DefaultPermissions.registerPermission(PREFIX + "command", "Allows the user to use all CloudPlane commands", PermissionDefault.FALSE, cloudplane);
+        DefaultPermissions.registerPermission(PREFIX + "command.fly", "Allows the user to use the fly command on himself", PermissionDefault.OP, commands);
+        DefaultPermissions.registerPermission(PREFIX + "command.fly.other", "Allows the user to use the fly command on others", PermissionDefault.OP, commands);
         commands.recalculatePermissibles();
 
         cloudplane.recalculatePermissibles();
