From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Wed, 28 Jul 2021 12:13:58 +0200
Subject: [PATCH] ported secure seed


diff --git a/src/main/java/org/bukkit/UnsafeValues.java b/src/main/java/org/bukkit/UnsafeValues.java
index b2502aaab690b1414a1adffdf64e5a5456feb99c..3719de8192f116bf79e9f8e3628d602ab602fb61 100644
--- a/src/main/java/org/bukkit/UnsafeValues.java
+++ b/src/main/java/org/bukkit/UnsafeValues.java
@@ -218,4 +218,12 @@ public interface UnsafeValues {
      */
     org.bukkit.entity.Entity deserializeEntity(byte[] data, org.bukkit.World world);
     // Purpur end
+
+    // CloudPlane start - port secure seed
+    /**
+     * Gets a random secure seed.
+     * @return a random secure seed
+     */
+    long[] randomSecureSeed();
+    // CloudPlane end
 }
diff --git a/src/main/java/org/bukkit/WorldCreator.java b/src/main/java/org/bukkit/WorldCreator.java
index d1eed20ea3b27f0800a08b42b62aac401bfe7347..ec2d7c6afe86ee31a8f3c3758229e571f1ee67e6 100644
--- a/src/main/java/org/bukkit/WorldCreator.java
+++ b/src/main/java/org/bukkit/WorldCreator.java
@@ -14,7 +14,7 @@ import org.jetbrains.annotations.Nullable;
 public class WorldCreator {
     private final NamespacedKey key; // Paper
     private final String name;
-    private long seed;
+    // private long seed; // CloudPlane - port secure seed
     private World.Environment environment = World.Environment.NORMAL;
     private ChunkGenerator generator = null;
     private BiomeProvider biomeProvider = null;
@@ -22,6 +22,7 @@ public class WorldCreator {
     private boolean generateStructures = true;
     private String generatorSettings = "";
     private boolean hardcore = false;
+    private long[] secureSeed; // CloudPlane - port secure seed
 
     /**
      * Creates an empty WorldCreationOptions for the given world name
@@ -44,8 +45,9 @@ public class WorldCreator {
             throw new IllegalArgumentException("World name and key cannot be null");
         }
         this.name = levelName;
-        this.seed = (new Random()).nextLong();
+        // this.seed = (new Random()).nextLong(); // CloudPlane - port secure seed
         this.key = worldKey;
+        this.secureSeed = Bukkit.getUnsafe().randomSecureSeed(); // CloudPlane - port secure seed
     }
 
     /**
@@ -107,13 +109,14 @@ public class WorldCreator {
             throw new IllegalArgumentException("World cannot be null");
         }
 
-        seed = world.getSeed();
+        // seed = world.getSeed(); // CloudPlane - port secure seed
         environment = world.getEnvironment();
         generator = world.getGenerator();
         biomeProvider = world.getBiomeProvider();
         type = world.getWorldType();
         generateStructures = world.canGenerateStructures();
         hardcore = world.isHardcore();
+        secureSeed = world.getSecureSeed(); // CloudPlane - port secure seed
 
         return this;
     }
@@ -130,7 +133,7 @@ public class WorldCreator {
             throw new IllegalArgumentException("Creator cannot be null");
         }
 
-        seed = creator.seed();
+        // seed = creator.seed(); // CloudPlane - port secure seed
         environment = creator.environment();
         generator = creator.generator();
         biomeProvider = creator.biomeProvider();
@@ -138,6 +141,7 @@ public class WorldCreator {
         generateStructures = creator.generateStructures();
         generatorSettings = creator.generatorSettings();
         hardcore = creator.hardcore();
+        secureSeed = creator.secureSeed(); // CloudPlane - port secure seed
 
         return this;
     }
@@ -158,7 +162,7 @@ public class WorldCreator {
      * @return World seed
      */
     public long seed() {
-        return seed;
+        return secureSeed[0]; // CloudPlane - port secure seed
     }
 
     /**
@@ -169,7 +173,8 @@ public class WorldCreator {
      */
     @NotNull
     public WorldCreator seed(long seed) {
-        this.seed = seed;
+        this.secureSeed = new long[this.secureSeed.length]; // CloudPlane - port secure seed
+        this.secureSeed[0] = seed; // CloudPlane - port secure seed
 
         return this;
     }
@@ -453,6 +458,29 @@ public class WorldCreator {
         return hardcore;
     }
 
+    // CloudPlane start - port secure seed
+    /**
+     * Sets the secure seed the world will have.
+     *
+     * @param secureSeed the secure seed for the world
+     * @return this object, for chaining
+     */
+    @NotNull
+    public WorldCreator secureSeed(long[] secureSeed) {
+        this.secureSeed = secureSeed;
+
+        return this;
+    }
+
+    /**
+     * Gets the secure seed the world will have.
+     * @return the secure seed the world will have
+     */
+    public long[] secureSeed() {
+        return secureSeed;
+    }
+    // CloudPlane end
+
     /**
      * Creates a world with the specified options.
      * <p>
diff --git a/src/main/java/org/bukkit/generator/WorldInfo.java b/src/main/java/org/bukkit/generator/WorldInfo.java
index 5bf00f46e8f3a80fec2a8a738e4d0d92461cfc92..e464ee038a82b96332f36402f658b2abf7487ddb 100644
--- a/src/main/java/org/bukkit/generator/WorldInfo.java
+++ b/src/main/java/org/bukkit/generator/WorldInfo.java
@@ -40,6 +40,14 @@ public interface WorldInfo {
      */
     long getSeed();
 
+    // CloudPlane start - port secure seed
+    /**
+     * Gets the secure seed for this world.
+     * @return This worlds secure seed
+     */
+    public long[] getSecureSeed();
+    // CloudPlane end
+
     /**
      * Gets the minimum height of this world.
      * <p>
