From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Fri, 2 Jul 2021 13:49:02 +0200
Subject: [PATCH] Added API methods for setting explosion resistance


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index b7c45aa5d5f79fd16e59401acfa9c2bed0a5d78c..575189f224c2113e831bf8973bae8e20ea58d2e3 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2138,4 +2138,14 @@ public final class Bukkit {
     public static Server.Spigot spigot() {
         return server.spigot();
     }
+
+    // CloudPlane start
+    public static float getExplosionResistance(@NotNull Material material) {
+        return server.getExplosionResistance(material);
+    }
+
+    public static void setExplosionResistance(@NotNull Material material, float explosionResistance) {
+        server.setExplosionResistance(material, explosionResistance);
+    }
+    // CloudPlane end
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index e5782bd5fe13639a9d5a29c22a72dedd8f785011..1e26c453ab5acce59a2799d3b71c2d85e1cea6c7 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1860,4 +1860,10 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     @NotNull
     io.papermc.paper.datapack.DatapackManager getDatapackManager();
     // Paper end
+
+    // CloudPlane start
+    float getExplosionResistance(@NotNull Material material);
+
+    void setExplosionResistance(@NotNull Material material, float explosionResistance);
+    // CloudPlane end
 }
