From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Fri, 2 Jul 2021 13:49:02 +0200
Subject: [PATCH] Added API methods for setting explosion resistance


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 53c4e5ca208ee17c7c244e416c537c7b63edf194..4afc5b79f67e22810eb636753664acea46094f4d 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2181,4 +2181,14 @@ public final class Bukkit {
     public static Server.Spigot spigot() {
         return server.spigot();
     }
+
+    // CloudPlane start
+    public static float getExplosionResistance(@NotNull Material material) {
+        return server.getExplosionResistance(material);
+    }
+
+    public static void setExplosionResistance(@NotNull Material material, float explosionResistance) {
+        server.setExplosionResistance(material, explosionResistance);
+    }
+    // CloudPlane end
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 86c3aef52623ae929b4293887db4f94cc46a58c9..98b2808a9959eb53dacbdbbf5792f583b3f838ac 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1895,4 +1895,10 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     @NotNull
     io.papermc.paper.datapack.DatapackManager getDatapackManager();
     // Paper end
+
+    // CloudPlane start
+    float getExplosionResistance(@NotNull Material material);
+
+    void setExplosionResistance(@NotNull Material material, float explosionResistance);
+    // CloudPlane end
 }
