From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Fri, 2 Jul 2021 13:49:02 +0200
Subject: [PATCH] api methods for setting explosion resistance


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 2e86aa13913553306512232a72497ec6b003260c..2dbeed9bb269bb61a225178ac111a6c31ff168ec 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2157,4 +2157,14 @@ public final class Bukkit {
         server.removeFuel(material);
     }
     // Purpur end
+
+    // CloudPlane start
+    public static float getExplosionResistance(@NotNull Material material) {
+        return server.getExplosionResistance(material);
+    }
+
+    public static void setExplosionResistance(@NotNull Material material, float explosionResistance) {
+        server.setExplosionResistance(material, explosionResistance);
+    }
+    // CloudPlane end
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 0177e40ed9d1e86c8a820a50c1fb019686bff990..04da407858d210ad66f9e1437f2e2732356b59c4 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1890,4 +1890,10 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
      */
     public void removeFuel(@NotNull Material material);
     // Purpur end
+
+    // CloudPlane start
+    float getExplosionResistance(@NotNull Material material);
+
+    void setExplosionResistance(@NotNull Material material, float explosionResistance);
+    // CloudPlane end
 }
