From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SuperKooks <superkooks@superkooks.com>
Date: Wed, 4 Aug 2021 18:18:52 +1000
Subject: [PATCH] Expose tracked entities


diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 18fef3648386853bb216c71305189716ddf0ea8d..f99a3a9566b33a48be79a5b5046347ed1198df28 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2450,4 +2450,14 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
      */
     public boolean hasFlyingFallDamage();
     // Purpur end
+
+    // CloudPlane start - expose tracked entities
+    /**
+     * Gets entities within this player's tracking range (that the player's client can "see")
+     * @param entityClass The class of entities to filter for
+     * @return The filtered et of entities
+     */
+    @NotNull
+    <T extends org.bukkit.entity.Entity> java.util.Collection<? extends T> getTrackedEntities(@NotNull Class<T> entityClass);
+    // CloudPlane end
 }
