From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <github@booky.tk>
Date: Sun, 27 Jun 2021 19:53:45 +0200
Subject: [PATCH] better configuration default options


diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
index f812f369e76fb4f46b20aeaac886032fbc1812e7..024f45fa15c71efb92e2c36958797498dec385ff 100644
--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
@@ -207,7 +207,7 @@ public class PaperConfig {
 
     public static String timingsServerName;
     private static void timings() {
-        boolean timings = getBoolean("timings.enabled", true);
+        boolean timings = getBoolean("timings.enabled", false); // CloudPlane - spark > timings
         // Airplane start
         boolean reallyEnableTimings = getBoolean("timings.really-enabled", false);
         if (timings && !reallyEnableTimings) {
diff --git a/src/main/java/net/pl3x/purpur/PurpurConfig.java b/src/main/java/net/pl3x/purpur/PurpurConfig.java
index 8244f362f551ed2bdef4df8e6b83cb176b4a4e59..a0fd4c709cc8e523c61b831bcbf277f6375432f4 100644
--- a/src/main/java/net/pl3x/purpur/PurpurConfig.java
+++ b/src/main/java/net/pl3x/purpur/PurpurConfig.java
@@ -362,8 +362,8 @@ public class PurpurConfig {
     public static String commandTPSBarTextColorMedium = "<gradient:#ffff55:#ffaa00><text></gradient>";
     public static String commandTPSBarTextColorLow = "<gradient:#ff5555:#aa0000><text></gradient>";
     public static int commandTPSBarTickInterval = 20;
-    public static boolean commandGamemodeRequiresPermission = false;
-    public static boolean hideHiddenPlayersFromEntitySelector = false;
+    public static boolean commandGamemodeRequiresPermission = true; // CloudPlane - better default option for gamemode requires specific permission
+    public static boolean hideHiddenPlayersFromEntitySelector = true; // CloudPlane - hidden players should not be accessible via entity selector
     public static String uptimeFormat = "<days><hours><minutes><seconds>";
     public static String uptimeDay = "%02d day, ";
     public static String uptimeDays = "%02d days, ";
