From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Sun, 27 Jun 2021 19:53:45 +0200
Subject: [PATCH] better configuration default options


diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
index e4e9be385c3ac048d7db374ee111de654dc3e3f0..28304903e468a063f97b4e790aab897002ae28fd 100644
--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
@@ -207,7 +207,7 @@ public class PaperConfig {
 
     public static String timingsServerName;
     private static void timings() {
-        boolean timings = getBoolean("timings.enabled", true);
+        boolean timings = getBoolean("timings.enabled", false); // CloudPlane - spark > timings
         // Airplane start
         boolean reallyEnableTimings = getBoolean("timings.really-enabled", false);
         if (timings && !reallyEnableTimings) {
diff --git a/src/main/java/net/pl3x/purpur/PurpurConfig.java b/src/main/java/net/pl3x/purpur/PurpurConfig.java
index 526a1032a2f61792c7f5791dbb3e9a7292fc186d..fcc1949184655bf4124ab18df8b871ea18f01fc0 100644
--- a/src/main/java/net/pl3x/purpur/PurpurConfig.java
+++ b/src/main/java/net/pl3x/purpur/PurpurConfig.java
@@ -374,8 +374,8 @@ public class PurpurConfig {
     public static BossBar.Color commandCompassBarProgressColor = BossBar.Color.BLUE;
     public static float commandCompassBarProgressPercent = 1.0F;
     public static int commandCompassBarTickInterval = 5;
-    public static boolean commandGamemodeRequiresPermission = false;
-    public static boolean hideHiddenPlayersFromEntitySelector = false;
+    public static boolean commandGamemodeRequiresPermission = true; // CloudPlane - better default option for gamemode requires specific permission
+    public static boolean hideHiddenPlayersFromEntitySelector = true; // CloudPlane - hidden players should not be accessible via entity selector
     public static String uptimeFormat = "<days><hours><minutes><seconds>";
     public static String uptimeDay = "%02d day, ";
     public static String uptimeDays = "%02d days, ";
