From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <github@booky.tk>
Date: Sun, 27 Jun 2021 19:53:45 +0200
Subject: [PATCH] better configuration default options


diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
index 9375c8471539ab6a84dbdef8bdc9c50b9e50fe09..68f5f97825c102bddf5ce350469275ae8a325450 100644
--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
@@ -207,7 +207,7 @@ public class PaperConfig {
 
     public static String timingsServerName;
     private static void timings() {
-        boolean timings = getBoolean("timings.enabled", true);
+        boolean timings = getBoolean("timings.enabled", false); // CloudPlane - spark > timings
         // Airplane start
         boolean reallyEnableTimings = getBoolean("timings.really-enabled", false);
         if (timings && !reallyEnableTimings) {
diff --git a/src/main/java/net/pl3x/purpur/PurpurConfig.java b/src/main/java/net/pl3x/purpur/PurpurConfig.java
index eaa0fcff033d59189475f7c04c5a3f2fc5421469..0d222595aca7ca185e2b15496b1936c8f4713491 100644
--- a/src/main/java/net/pl3x/purpur/PurpurConfig.java
+++ b/src/main/java/net/pl3x/purpur/PurpurConfig.java
@@ -312,8 +312,8 @@ public class PurpurConfig {
         disableGiveCommandDrops = getBoolean("settings.disable-give-dropping", disableGiveCommandDrops);
     }
 
-    public static boolean commandGamemodeRequiresPermission = false;
-    public static boolean hideHiddenPlayersFromEntitySelector = false;
+    public static boolean commandGamemodeRequiresPermission = true; // CloudPlane - better default option for gamemode requires specific permission
+    public static boolean hideHiddenPlayersFromEntitySelector = true; // CloudPlane - hidden players should not be accessible via entity selector
     private static void commandSettings() {
         commandGamemodeRequiresPermission = getBoolean("settings.command.gamemode.requires-specific-permission", commandGamemodeRequiresPermission);
         hideHiddenPlayersFromEntitySelector = getBoolean("settings.command.hide-hidden-players-from-entity-selector", hideHiddenPlayersFromEntitySelector);
