From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <github@booky.tk>
Date: Sun, 27 Jun 2021 19:53:45 +0200
Subject: [PATCH] better configuration default options


diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
index 9375c8471539ab6a84dbdef8bdc9c50b9e50fe09..92dea2ff93213c813202807bbce2b03f0bd8b691 100644
--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
@@ -207,7 +207,7 @@ public class PaperConfig {
 
     public static String timingsServerName;
     private static void timings() {
-        boolean timings = getBoolean("timings.enabled", true);
+        boolean timings = getBoolean("timings.enabled", false); // CloudPlane - spark > timings
         // Airplane start
         boolean reallyEnableTimings = getBoolean("timings.really-enabled", false);
         if (timings && !reallyEnableTimings) {
diff --git a/src/main/java/net/pl3x/purpur/PurpurConfig.java b/src/main/java/net/pl3x/purpur/PurpurConfig.java
index 9592a9e6743e7fe0250de16680dd0d3e2e5e7776..879e5480a5b75bf5e14e604aebd8fed47973704b 100644
--- a/src/main/java/net/pl3x/purpur/PurpurConfig.java
+++ b/src/main/java/net/pl3x/purpur/PurpurConfig.java
@@ -344,8 +344,8 @@ public class PurpurConfig {
         disableGiveCommandDrops = getBoolean("settings.disable-give-dropping", disableGiveCommandDrops);
     }
 
-    public static boolean commandGamemodeRequiresPermission = false;
-    public static boolean hideHiddenPlayersFromEntitySelector = false;
+    public static boolean commandGamemodeRequiresPermission = true; // CloudPlane - better default option for gamemode requires specific permission
+    public static boolean hideHiddenPlayersFromEntitySelector = true; // CloudPlane - hidden players should not be accessible via entity selector
     public static String uptimeFormat = "<days><hours><minutes><seconds>";
     public static String uptimeDay = "%02d day, ";
     public static String uptimeDays = "%02d days, ";
