From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Fri, 25 Jun 2021 22:20:11 +0200
Subject: [PATCH] added fly command using brigadier


diff --git a/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java b/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
index 7ea3ccc82b38bbbd74ce338b55237b74c3769cd2..972c1c1c73d147b56e6cf6b509086370aa81fcea 100644
--- a/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
+++ b/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
@@ -63,6 +63,8 @@ public class CloudPlaneConfig {
     public static void registerCommands(CommandDispatcher<CommandSourceStack> dispatcher) {
         commands.clear();
 
+        commands.add(new tk.booky.cloudplane.commands.FlyCommand());
+
         commands.forEach(command -> command.register(dispatcher));
     }
 
diff --git a/src/main/java/tk/booky/cloudplane/commands/FlyCommand.java b/src/main/java/tk/booky/cloudplane/commands/FlyCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..79b9c4067f4f42d377b083f8985c7a7e99461f95
--- /dev/null
+++ b/src/main/java/tk/booky/cloudplane/commands/FlyCommand.java
@@ -0,0 +1,58 @@
+package tk.booky.cloudplane.commands;
+// Created by booky10 in CloudPlane (22:08 25.06.21)
+
+import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.arguments.BoolArgumentType;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.commands.Commands;
+import net.minecraft.commands.arguments.EntityArgument;
+import net.minecraft.server.level.ServerPlayer;
+import org.bukkit.entity.Player;
+
+import java.util.Collection;
+import java.util.Collections;
+
+public class FlyCommand implements CloudPlaneCommand {
+
+    @Override
+    public void register(CommandDispatcher<CommandSourceStack> dispatcher) {
+        dispatcher.register(Commands.literal("fly")
+                .requires((listener) -> listener.hasPermission(2, "bukkit.command.fly"))
+                .executes((context) -> execute(context.getSource(), Collections.singleton(context.getSource().getPlayerOrException())))
+                .then(Commands.argument("allow", BoolArgumentType.bool())
+                        .executes((context) -> execute(context.getSource(), Collections.singleton(context.getSource().getPlayerOrException()), BoolArgumentType.getBool(context, "allow")))
+                        .then(Commands.argument("targets", EntityArgument.players())
+                                .requires((listener) -> listener.hasPermission(3, "bukkit.command.fly.other"))
+                                .executes((context) -> execute(context.getSource(), EntityArgument.getPlayers(context, "targets"), BoolArgumentType.getBool(context, "allow")))
+                        )
+                )
+                .then(Commands.argument("targets", EntityArgument.players())
+                        .requires((listener) -> listener.hasPermission(3, "bukkit.command.fly.other"))
+                        .executes((context) -> execute(context.getSource(), EntityArgument.getPlayers(context, "targets")))
+                )
+        );
+    }
+
+    private int execute(CommandSourceStack sender, Collection<ServerPlayer> targets, boolean allowFlight) {
+        for (ServerPlayer player : targets) {
+            Player bukkit = player.getBukkitEntity();
+
+            bukkit.setAllowFlight(allowFlight);
+            sender.sendSuccess(player.getGameProfile().getName() + "'s fly mode was " + (allowFlight ? "enabled" : "disabled"), true);
+        }
+
+        return targets.size();
+    }
+
+    private int execute(CommandSourceStack sender, Collection<ServerPlayer> targets) {
+        for (ServerPlayer player : targets) {
+            Player bukkit = player.getBukkitEntity();
+            boolean allowFlight = !bukkit.getAllowFlight();
+
+            bukkit.setAllowFlight(allowFlight);
+            sender.sendSuccess(player.getGameProfile().getName() + "'s fly mode was " + (allowFlight ? "enabled" : "disabled"), true);
+        }
+
+        return targets.size();
+    }
+}
