From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Sun, 27 Jun 2021 21:06:03 +0200
Subject: [PATCH] added configurable sleep message


diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index bf91d6c0d67aa72a6272dc878b07e292386f94f2..7a3288f9a83ae36c4947f56265d487a2ab3e0e62 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -735,6 +735,11 @@ public class ServerLevel extends Level implements WorldGenLevel {
             if (this.getGameRules().getBoolean(GameRules.RULE_WEATHER_CYCLE)) {
                 this.stopWeather();
             }
+            // CloudPlane start - configurable sleeping over message
+            if (!org.apache.commons.lang3.StringUtils.isEmpty(tk.booky.cloudplane.CloudPlaneConfig.sleepMessage)) {
+                Bukkit.broadcast(net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacyAmpersand().deserialize(tk.booky.cloudplane.CloudPlaneConfig.sleepMessage));
+            }
+            // CloudPlane end
         }
 
         this.updateSkyBrightness();
diff --git a/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java b/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
index a2ac3976c168c32c7669b3a49884a3ef5d57b785..41577145f2d37635401fcc8355061ff604139b62 100644
--- a/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
+++ b/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
@@ -126,4 +126,15 @@ public class CloudPlaneConfig {
         config.addDefault(path, def);
         return config.getString(path, config.getString(path));
     }
+
+    public static String sleepMessage = "The night has been skipped!";
+    private static void sleepMessage() {
+        if (currentVersion == 1) {
+            sleepMessage = getString("settings.sleeping.message", sleepMessage);
+            set("settings.sleep-message", sleepMessage);
+            set("settings.sleeping", null);
+        }
+
+        sleepMessage = getString("settings.sleep-message", sleepMessage);
+    }
 }
