From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <github@booky.tk>
Date: Sun, 27 Jun 2021 21:06:03 +0200
Subject: [PATCH] added configurable sleeping over message


diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index 4caa8a065d33247e13baa2eea633cd69ce83caf7..589f02230b7442535eaf992aa1d4bb16d32a9cae 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -800,6 +800,11 @@ public class ServerLevel extends net.minecraft.world.level.Level implements Worl
             if (this.getGameRules().getBoolean(GameRules.RULE_WEATHER_CYCLE)) {
                 this.stopWeather();
             }
+            // CloudPlane start - configurable sleeping over message
+            if (!org.apache.commons.lang3.StringUtils.isEmpty(tk.booky.cloudplane.CloudPlaneConfig.sleepMessage)) {
+                Bukkit.broadcast(net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacyAmpersand().deserialize(tk.booky.cloudplane.CloudPlaneConfig.sleepMessage));
+            }
+            // CloudPlane end
         }
 
         this.updateSkyBrightness();
diff --git a/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java b/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
index 972c1c1c73d147b56e6cf6b509086370aa81fcea..d9955a86ba2b6a5015e95f5b07258f1995d8f834 100644
--- a/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
+++ b/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
@@ -126,4 +126,9 @@ public class CloudPlaneConfig {
         config.addDefault(path, def);
         return config.getString(path, config.getString(path));
     }
+
+    public static String sleepMessage = "The night has been skipped!";
+    private static void sleeping() {
+        sleepMessage = getString("settings.sleeping.message", sleepMessage);
+    }
 }
