From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Thu, 1 Jul 2021 21:58:19 +0200
Subject: [PATCH] added invisible item frames


diff --git a/src/main/java/net/minecraft/world/entity/decoration/ItemFrame.java b/src/main/java/net/minecraft/world/entity/decoration/ItemFrame.java
index 90f1ecd97eef93316f61bf5d761d33c313e1a077..917c2ca3c32ff548afe05853346dce16f9ccca42 100644
--- a/src/main/java/net/minecraft/world/entity/decoration/ItemFrame.java
+++ b/src/main/java/net/minecraft/world/entity/decoration/ItemFrame.java
@@ -187,6 +187,12 @@ public class ItemFrame extends HangingEntity {
                 // CraftBukkit end
                 this.dropItem(source.getEntity(), false);
                 this.playSound(this.getRemoveItemSound(), 1.0F, 1.0F);
+                // CloudPlane start - invisible item frames
+                if (noCulling) {
+                    setInvisible(false);
+                    setGlowingTag(true);
+                }
+                // CloudPlane end
             }
 
             return true;
@@ -434,6 +440,12 @@ public class ItemFrame extends HangingEntity {
                         }
                     }
 
+                    // CloudPlane start - invisible item frames
+                    if (noCulling) {
+                        setGlowingTag(false);
+                        setInvisible(true);
+                    }
+                    // CloudPlane end
                     this.setItem(itemstack);
                     if (!player.getAbilities().instabuild) {
                         itemstack.shrink(1);
diff --git a/src/main/java/net/minecraft/world/item/HangingEntityItem.java b/src/main/java/net/minecraft/world/item/HangingEntityItem.java
index 8cdede6a9698d6c99b4f08edc2a3403fec854f66..1d38fe546cf48dc35ef22c35b33365ba2fce7e66 100644
--- a/src/main/java/net/minecraft/world/item/HangingEntityItem.java
+++ b/src/main/java/net/minecraft/world/item/HangingEntityItem.java
@@ -46,6 +46,14 @@ public class HangingEntityItem extends Item {
                 object = new Painting(world, blockposition1, enumdirection);
             } else if (this.type == EntityType.ITEM_FRAME) {
                 object = new ItemFrame(world, blockposition1, enumdirection);
+                // CloudPlane start - invisible item frames
+                if (itemstack.tag != null && !itemstack.tag.isEmpty()) {
+                    if (itemstack.tag.getInt("CustomModelData") == 1) {
+                        object.setGlowingTag(true);
+                        object.noCulling = true;
+                    }
+                }
+                // CloudPlane end
             } else {
                 if (this.type != EntityType.GLOW_ITEM_FRAME) {
                     return InteractionResult.sidedSuccess(world.isClientSide);
diff --git a/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java b/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
index f077828be3b355c10dc6e06488c3531e1521e472..7c5cbfd668b5c56dc0ee7d55881592970ad93744 100644
--- a/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
+++ b/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
@@ -71,6 +71,8 @@ public class CloudPlaneConfig {
     public static void registerRecipes(Server server) {
         recipes.clear();
 
+        recipes.add(tk.booky.cloudplane.recipes.InvisibleItemFrameRecipe.get());
+
         recipes.forEach(server::addRecipe);
     }
 
diff --git a/src/main/java/tk/booky/cloudplane/recipes/InvisibleItemFrameRecipe.java b/src/main/java/tk/booky/cloudplane/recipes/InvisibleItemFrameRecipe.java
new file mode 100644
index 0000000000000000000000000000000000000000..106c33201526f4c9676f3d6ee3ac51c9fe5ef35d
--- /dev/null
+++ b/src/main/java/tk/booky/cloudplane/recipes/InvisibleItemFrameRecipe.java
@@ -0,0 +1,28 @@
+package tk.booky.cloudplane.recipes;
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.TextDecoration;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.Recipe;
+import org.bukkit.inventory.ShapedRecipe;
+import org.bukkit.inventory.meta.ItemMeta;
+
+public class InvisibleItemFrameRecipe {
+
+    private static final Component NAME = Component.text("Invisible Item Frame").decoration(TextDecoration.ITALIC, false);
+    private static final NamespacedKey KEY = NamespacedKey.minecraft("invisible_item_frame");
+    private static final String[] SHAPE = new String[]{"###", "#+#", "###"};
+
+    public static Recipe get() {
+        ItemStack item = new ItemStack(Material.ITEM_FRAME);
+        ItemMeta meta = item.getItemMeta();
+
+        meta.setCustomModelData(1);
+        meta.displayName(NAME);
+
+        item.setItemMeta(meta);
+        return new ShapedRecipe(KEY, item).shape(SHAPE).setIngredient('#', Material.GLASS_PANE).setIngredient('+', Material.ITEM_FRAME);
+    }
+}
