From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <github@booky.tk>
Date: Thu, 1 Jul 2021 21:58:19 +0200
Subject: [PATCH] added invisible item frames


diff --git a/src/main/java/net/minecraft/world/item/HangingEntityItem.java b/src/main/java/net/minecraft/world/item/HangingEntityItem.java
index ddd50db8bb92c147d7c1eef4d1df3ac53e2d3e4b..ffad816ea70b272aa6b9fae1cfa0f0133d89fb93 100644
--- a/src/main/java/net/minecraft/world/item/HangingEntityItem.java
+++ b/src/main/java/net/minecraft/world/item/HangingEntityItem.java
@@ -46,6 +46,11 @@ public class HangingEntityItem extends Item {
                 object = new Painting(world, blockposition1, enumdirection);
             } else if (this.type == EntityType.ITEM_FRAME) {
                 object = new ItemFrame(world, blockposition1, enumdirection);
+                // CloudPlane start - invisible item frames
+                if (itemstack.tag != null && !itemstack.tag.isEmpty()) {
+                    object.setInvisible(itemstack.tag.getInt("CustomModelData") == 1);
+                }
+                // CloudPlane end
             } else {
                 if (this.type != EntityType.GLOW_ITEM_FRAME) {
                     return InteractionResult.sidedSuccess(world.isClientSide);
diff --git a/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java b/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
index acdf5ea416e60adfdc191572190ea398700fc660..8d1128fb029ac686513a0f25fe86f7071269afb8 100644
--- a/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
+++ b/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
@@ -72,6 +72,8 @@ public class CloudPlaneConfig {
     public static void registerRecipes(Server server) {
         recipes.clear();
 
+        recipes.add(tk.booky.cloudplane.recipes.InvisibleItemFrameRecipe.get());
+
         recipes.forEach(server::addRecipe);
     }
 
diff --git a/src/main/java/tk/booky/cloudplane/recipes/InvisibleItemFrameRecipe.java b/src/main/java/tk/booky/cloudplane/recipes/InvisibleItemFrameRecipe.java
new file mode 100644
index 0000000000000000000000000000000000000000..b6ea11001e70bb4a931d7d761fa7050192717c10
--- /dev/null
+++ b/src/main/java/tk/booky/cloudplane/recipes/InvisibleItemFrameRecipe.java
@@ -0,0 +1,29 @@
+package tk.booky.cloudplane.recipes;
+// Created by booky10 in CloudPlane (21:50 01.07.21)
+
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.TextDecoration;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.Recipe;
+import org.bukkit.inventory.ShapedRecipe;
+import org.bukkit.inventory.meta.ItemMeta;
+
+public class InvisibleItemFrameRecipe {
+
+    private static final Component NAME = Component.text("Invisible Item Frame").decoration(TextDecoration.ITALIC, false);
+    private static final NamespacedKey KEY = NamespacedKey.minecraft("invisible_item_frame");
+    private static final String[] SHAPE = new String[]{"###", "#+#", "###"};
+
+    public static Recipe get() {
+        ItemStack item = new ItemStack(Material.ITEM_FRAME);
+        ItemMeta meta = item.getItemMeta();
+
+        meta.setCustomModelData(1);
+        meta.displayName(NAME);
+
+        item.setItemMeta(meta);
+        return new ShapedRecipe(KEY, item).shape(SHAPE).setIngredient('#', Material.GLASS_PANE).setIngredient('+', Material.ITEM_FRAME);
+    }
+}
\ No newline at end of file
