From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Fri, 2 Jul 2021 13:49:01 +0200
Subject: [PATCH] Added API methods for setting explosion resistance


diff --git a/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java b/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java
index b2dd2a211bdd1441eae67dd6bd93691964e07620..2a26804eb75d7595dc4a01a5f61b4e907fb7f59b 100644
--- a/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java
+++ b/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java
@@ -71,7 +71,7 @@ public abstract class BlockBehaviour {
     protected static final Direction[] UPDATE_SHAPE_ORDER = new Direction[]{Direction.WEST, Direction.EAST, Direction.NORTH, Direction.SOUTH, Direction.DOWN, Direction.UP};
     public final Material material; // Purpur - protected -> public
     public final boolean hasCollision;
-    protected final float explosionResistance;
+    public float explosionResistance; // CloudPlane - protected final -> public
     protected final boolean isRandomlyTicking;
     protected final SoundType soundType;
     protected final float friction;
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index ad143f05893fc6e5a60182501e939228675c9dc5..9444256f266e4584f98beebecda882958e991c51 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2856,4 +2856,16 @@ public final class CraftServer implements Server {
         return getServer().lagging;
     }
     // Purpur end
+
+    // CloudPlane start
+    @Override
+    public float getExplosionResistance(org.bukkit.Material material) {
+        return CraftMagicNumbers.getBlock(material).explosionResistance;
+    }
+
+    @Override
+    public void setExplosionResistance(org.bukkit.Material material, float explosionResistance) {
+        CraftMagicNumbers.getBlock(material).explosionResistance = explosionResistance;
+    }
+    // CloudPlane end
 }
