From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Fri, 2 Jul 2021 13:49:01 +0200
Subject: [PATCH] api methods for setting explosion resistance


diff --git a/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java b/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java
index e1aba175a74e73db0a8a11e8d2df35c45adfd3fa..534b13346e0fd6cb3c51db11235205b9f339e6b9 100644
--- a/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java
+++ b/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java
@@ -72,7 +72,7 @@ public abstract class BlockBehaviour {
     protected static final Direction[] UPDATE_SHAPE_ORDER = new Direction[]{Direction.WEST, Direction.EAST, Direction.NORTH, Direction.SOUTH, Direction.DOWN, Direction.UP};
     public final Material material; // Purpur - protected -> public
     protected final boolean hasCollision;
-    protected final float explosionResistance;
+    public float explosionResistance; // CloudPlane - protected final -> public
     protected final boolean isRandomlyTicking;
     protected final SoundType soundType;
     protected final float friction;
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index f4402cf3feb4c969d67541db7d75bc8021835dbd..4e3c4b304ef5ea7eccb61fce939fd0ff04e0a1b8 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2717,4 +2717,16 @@ public final class CraftServer implements Server {
         return getServer().lagging;
     }
     // Purpur end
+
+    // CloudPlane start
+    @Override
+    public float getExplosionResistance(org.bukkit.Material material) {
+        return CraftMagicNumbers.getBlock(material).explosionResistance;
+    }
+
+    @Override
+    public void setExplosionResistance(org.bukkit.Material material, float explosionResistance) {
+        CraftMagicNumbers.getBlock(material).explosionResistance = explosionResistance;
+    }
+    // CloudPlane end
 }
