From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Fri, 2 Jul 2021 13:49:01 +0200
Subject: [PATCH] Added API methods for setting explosion resistance


diff --git a/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java b/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java
index 05c46f3b3bce5225b819d86e6e06729a5093e092..0c80af638cc99d02625caa4880033a762b8608b5 100644
--- a/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java
+++ b/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java
@@ -71,7 +71,7 @@ public abstract class BlockBehaviour {
     protected static final Direction[] UPDATE_SHAPE_ORDER = new Direction[]{Direction.WEST, Direction.EAST, Direction.NORTH, Direction.SOUTH, Direction.DOWN, Direction.UP};
     protected final Material material;
     public final boolean hasCollision;
-    protected final float explosionResistance;
+    public float explosionResistance; // CloudPlane - protected final -> public
     protected final boolean isRandomlyTicking;
     protected final SoundType soundType;
     protected final float friction;
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index f1bd9ab21d717c7086a04215ee24159220e668dd..280ae71fd0ba17a4e12a83f105f4f9d00dcbae64 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2808,4 +2808,16 @@ public final class CraftServer implements Server {
     }
 
     // Paper end
+
+    // CloudPlane start
+    @Override
+    public float getExplosionResistance(org.bukkit.Material material) {
+        return CraftMagicNumbers.getBlock(material).explosionResistance;
+    }
+
+    @Override
+    public void setExplosionResistance(org.bukkit.Material material, float explosionResistance) {
+        CraftMagicNumbers.getBlock(material).explosionResistance = explosionResistance;
+    }
+    // CloudPlane end
 }
