From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <github@booky.tk>
Date: Fri, 2 Jul 2021 13:49:01 +0200
Subject: [PATCH] api methods for setting explosion resistance


diff --git a/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java b/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java
index f34973be478de4f088a0593b45bd89e558a13609..b157bf6af3d2d366a48a5991c14e335d6da7954f 100644
--- a/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java
+++ b/src/main/java/net/minecraft/world/level/block/state/BlockBehaviour.java
@@ -72,7 +72,7 @@ public abstract class BlockBehaviour {
     protected static final Direction[] UPDATE_SHAPE_ORDER = new Direction[]{Direction.WEST, Direction.EAST, Direction.NORTH, Direction.SOUTH, Direction.DOWN, Direction.UP};
     public final Material material; // Purpur - protected -> public
     protected final boolean hasCollision;
-    protected final float explosionResistance;
+    public float explosionResistance; // CloudPlane - protected final -> public
     protected final boolean isRandomlyTicking;
     protected final SoundType soundType;
     protected final float friction;
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 0db9337000ea123215792292ff5748d58efe6b69..b133518b63db3067a55d7909876ddcf5bcf4a927 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2658,4 +2658,16 @@ public final class CraftServer implements Server {
         return getServer().lagging;
     }
     // Purpur end
+
+    // CloudPlane start
+    @Override
+    public float getExplosionResistance(org.bukkit.Material material) {
+        return CraftMagicNumbers.getBlock(material).explosionResistance;
+    }
+
+    @Override
+    public void setExplosionResistance(org.bukkit.Material material, float explosionResistance) {
+        CraftMagicNumbers.getBlock(material).explosionResistance = explosionResistance;
+    }
+    // CloudPlane end
 }
