From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <github@booky.tk>
Date: Sun, 4 Jul 2021 22:58:51 +0200
Subject: [PATCH] use 1.17 mojang api


diff --git a/build.gradle.kts b/build.gradle.kts
index 1bd26e1b7dc90dad31512e5115d66a1f5218eb5f..1fb22e20ab5e39e4f0a0dd6604f5b18ba5503d10 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -36,7 +36,7 @@ repositories {
 
 dependencies {
     implementation(project(":CloudPlane-API")) // CloudPlane // Airplane // Tuinity
-    implementation("com.destroystokyo.paper:paper-mojangapi:1.16.5-R0.1-SNAPSHOT") // Tuinity
+    implementation("io.papermc.paper:paper-mojangapi:1.17-R0.1-SNAPSHOT") // CloudPlane - use 1.17 mojang api // Tuinity
     // Paper start
     implementation("org.jline:jline-terminal-jansi:3.12.1")
     implementation("net.minecrell:terminalconsoleappender:1.2.0")
