From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Mon, 5 Jul 2021 19:58:07 +0200
Subject: [PATCH] configurable ban-screen date format


diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index e37f80065134fd7929ca6cf0838b2bc5b67cbc7b..74265a7dd3622149fa6ecdc0ac6e677b4ed92e78 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -745,7 +745,7 @@ public abstract class PlayerList {
 
             chatmessage = new TranslatableComponent("multiplayer.disconnect.banned.reason", new Object[]{gameprofilebanentry.getReason()});
             if (gameprofilebanentry.getExpires() != null) {
-                chatmessage.append(new TranslatableComponent("multiplayer.disconnect.banned.expiration", new Object[]{PlayerList.BAN_DATE_FORMAT.format(gameprofilebanentry.getExpires())}));
+                chatmessage.append(new TranslatableComponent("multiplayer.disconnect.banned.expiration", new Object[]{tk.booky.cloudplane.CloudPlaneConfig.expireDateFormat.format(gameprofilebanentry.getExpires())})); // CloudPlane - configurable date format
             }
 
             // return chatmessage;
@@ -758,7 +758,7 @@ public abstract class PlayerList {
 
             chatmessage = new TranslatableComponent("multiplayer.disconnect.banned_ip.reason", new Object[]{ipbanentry.getReason()});
             if (ipbanentry.getExpires() != null) {
-                chatmessage.append(new TranslatableComponent("multiplayer.disconnect.banned_ip.expiration", new Object[]{PlayerList.BAN_DATE_FORMAT.format(ipbanentry.getExpires())}));
+                chatmessage.append(new TranslatableComponent("multiplayer.disconnect.banned_ip.expiration", new Object[]{tk.booky.cloudplane.CloudPlaneConfig.expireDateFormat.format(ipbanentry.getExpires())})); // CloudPlane - configurable date format
             }
 
             // return chatmessage;
diff --git a/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java b/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
index e437153fabb826421b76efc712eaf2baae25addf..c38f6f76ca2520e67599c5e55dea03783af184f1 100644
--- a/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
+++ b/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
@@ -136,4 +136,9 @@ public class CloudPlaneConfig {
     private static void sleeping() {
         sleepMessage = getString("settings.sleeping.message", sleepMessage);
     }
+
+    public static java.text.SimpleDateFormat expireDateFormat = (java.text.SimpleDateFormat) net.minecraft.server.players.PlayerList.BAN_DATE_FORMAT.clone();
+    private static void banning() {
+        expireDateFormat = new java.text.SimpleDateFormat(getString("settings.banning.date-format", expireDateFormat.toPattern()));
+    }
 }
diff --git a/src/main/java/tk/booky/cloudplane/commands/TempBanCommand.java b/src/main/java/tk/booky/cloudplane/commands/TempBanCommand.java
index 7a68603c55ab2f0d91ec3fbbd28d4ab577f9fc14..0466574e70fb6be6cb95207a147e8eb1a02398eb 100644
--- a/src/main/java/tk/booky/cloudplane/commands/TempBanCommand.java
+++ b/src/main/java/tk/booky/cloudplane/commands/TempBanCommand.java
@@ -68,7 +68,7 @@ public class TempBanCommand implements CloudPlaneCommand {
 
             if (target == null) continue;
             BaseComponent component = new TranslatableComponent("multiplayer.disconnect.banned");
-            component.append(new TranslatableComponent("multiplayer.disconnect.banned.expiration", PlayerList.BAN_DATE_FORMAT.format(expirationDate)));
+            component.append(new TranslatableComponent("multiplayer.disconnect.banned.expiration", tk.booky.cloudplane.CloudPlaneConfig.expireDateFormat.format(expirationDate))); // CloudPlane - configurable date format
             target.connection.disconnect(component, PlayerKickEvent.Cause.BANNED);
         }
 
