From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <github@booky.tk>
Date: Sun, 4 Jul 2021 23:15:59 +0200
Subject: [PATCH] revert tuinity changes in BlockPos.java, somehow causes
 errors


diff --git a/src/main/java/net/minecraft/core/BlockPos.java b/src/main/java/net/minecraft/core/BlockPos.java
index 177134c0f232e04cb081c3182340f764fbaf4b8e..d0791603ca9ac0530b74a5b5ea6afcb5001946f9 100644
--- a/src/main/java/net/minecraft/core/BlockPos.java
+++ b/src/main/java/net/minecraft/core/BlockPos.java
@@ -484,9 +484,9 @@ public class BlockPos extends Vec3i {
         }
 
         public BlockPos.MutableBlockPos set(int x, int y, int z) {
-            ((Vec3i) this).x = x; // Tuinity - force inline
-            ((Vec3i) this).y = y; // Tuinity - force inline
-            ((Vec3i) this).z = z; // Tuinity - force inline
+            ((Vec3i) this).x = x; // Tuinity - force inline // CloudPlane - Revert
+            ((Vec3i) this).y = y; // Tuinity - force inline // CloudPlane - Revert
+            ((Vec3i) this).z = z; // Tuinity - force inline // CloudPlane - Revert
             return this;
         }
 
@@ -550,19 +550,19 @@ public class BlockPos extends Vec3i {
         // Paper start - comment out useless overrides @Override - TODO figure out why this is suddenly important to keep
         @Override
         public BlockPos.MutableBlockPos setX(int i) {
-            ((Vec3i) this).x = i; // Tuinity
+            ((Vec3i) this).x = i; // Tuinity // CloudPlane - Revert
             return this;
         }
 
         @Override
         public BlockPos.MutableBlockPos setY(int i) {
-            ((Vec3i) this).y = i; // Tuinity
+            ((Vec3i) this).y = i; // Tuinity // CloudPlane - Revert
             return this;
         }
 
         @Override
         public BlockPos.MutableBlockPos setZ(int i) {
-            ((Vec3i) this).z = i; // Tuinity
+            ((Vec3i) this).z = i; // Tuinity // CloudPlane - Revert
             return this;
         }
         // Paper end
