From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Sun, 12 Sep 2021 12:50:09 +0200
Subject: [PATCH] Allow changing pufferfish config file location


diff --git a/src/main/java/gg/pufferfish/pufferfish/PufferfishCommand.java b/src/main/java/gg/pufferfish/pufferfish/PufferfishCommand.java
index 020368da69b9a492155f6de6297f74732f4ab6ea..4a5ad92c7989d4f85eea3f2908732fb527f7868f 100644
--- a/src/main/java/gg/pufferfish/pufferfish/PufferfishCommand.java
+++ b/src/main/java/gg/pufferfish/pufferfish/PufferfishCommand.java
@@ -50,7 +50,7 @@ public class PufferfishCommand extends Command {
         if (args[0].equalsIgnoreCase("reload")) {
             MinecraftServer console = MinecraftServer.getServer();
             try {
-                PufferfishConfig.load();
+                PufferfishConfig.load((java.io.File) console.options.valueOf("pufferfish-settings")); // CloudPlane
             } catch (IOException e) {
                 sender.sendMessage(Component.text("Failed to reload.", NamedTextColor.RED));
                 e.printStackTrace();
diff --git a/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java b/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
index 56330536c52fa327ef89d7a08e72557c6633c8bb..2a58384d16ed1e9c2b10d8b37c26228a17e81a9e 100644
--- a/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
+++ b/src/main/java/gg/pufferfish/pufferfish/PufferfishConfig.java
@@ -54,8 +54,8 @@ public class PufferfishConfig {
 		return updates;
 	}
 	
-	public static void load() throws IOException {
-		File configFile = new File("pufferfish.yml");
+	public static void load(File configFile) throws IOException { // CloudPlane
+		// File configFile = new File("pufferfish.yml"); // CloudPlane
 		
 		if (configFile.exists()) {
 			try {
diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index a4c9180037973b73c06340cb0d93093bb1401184..3171e4868e464f521a5c2327d9d64b60dd50c1be 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -243,7 +243,7 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         io.papermc.paper.util.ObfHelper.INSTANCE.getClass(); // load mappings for stacktrace deobf and etc.
         io.papermc.paper.brigadier.PaperBrigadierProviderImpl.INSTANCE.getClass(); // init PaperBrigadierProvider
         // Paper end
-        gg.pufferfish.pufferfish.PufferfishConfig.load(); // Pufferfish
+        gg.pufferfish.pufferfish.PufferfishConfig.load((java.io.File) options.valueOf("pufferfish-settings")); // Pufferfish // CloudPlane
         gg.pufferfish.pufferfish.PufferfishCommand.init(); // Pufferfish
 
         this.setPvpAllowed(dedicatedserverproperties.pvp);
diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index 8818d26950b9f562d5326fe241657f28a28b7f50..97c796d30891fed19a35c698aa7052e1d82f9bee 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -159,6 +159,12 @@ public class Main {
                         .ofType(File.class)
                         .defaultsTo(new File("cloudplane.yml"))
                         .describedAs("Yml file");
+
+                acceptsAll(asList("pufferfish", "pufferfish-settings"), "File for pufferfish settings")
+                        .withRequiredArg()
+                        .ofType(File.class)
+                        .defaultsTo(new File("pufferfish.yml"))
+                        .describedAs("Yml file");
                 // CloudPlane end
 
                 // Paper start
