From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SuperKooks <superkooks@superkooks.com>
Date: Wed, 4 Aug 2021 18:05:01 +1000
Subject: [PATCH] Expose tracked entities


diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
index d13b53c5497cece27ce06d43401e4f3b85652ceb..01af7b6fc00366818b8c73a2490fdc0fa52ba8fd 100644
--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
+++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
@@ -2243,7 +2243,7 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
 
         final ServerEntity serverEntity;
         final Entity entity;
-        private final int range;
+        public final int range; // CloudPlane - private --> public
         SectionPos lastSectionPos;
         public final Set<ServerPlayerConnection> seenBy = new ReferenceOpenHashSet<>(); // Paper - optimise map impl
 
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 603e9234294c01f1cde4bfc0b0053e69b99861d9..23953857622a2483d59d8f68243d80ff1d40d2f9 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2714,4 +2714,12 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         return this.spigot;
     }
     // Spigot end
+
+    // CloudPlane start - expose tracked entities
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T extends org.bukkit.entity.Entity> Collection<? extends T> getTrackedEntities(Class<T> entityClass) {
+        return entityClass == Player.class ? (Set<T>) getTrackedPlayers() : getWorld().getNearbyEntitiesByType(entityClass, getLocation(), entity.tracker.range);
+    }
+    // CloudPlane end
 }
