From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SuperKooks <superkooks@superkooks.com>
Date: Wed, 4 Aug 2021 18:05:01 +1000
Subject: [PATCH] Expose tracked entities


diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
index 862b4fccd741cb9de7a00a45f9999a7f9fe7a157..ea046b203a1a0b5d492ff97ffc4aa8b9cb4ecede 100644
--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
+++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
@@ -2253,7 +2253,7 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
 
         final ServerEntity serverEntity;
         final Entity entity;
-        private final int range;
+        public final int range; // CloudPlane - private --> public
         SectionPos lastSectionPos;
         public final Set<ServerPlayerConnection> seenBy = new ReferenceOpenHashSet<>(); // Paper - optimise map impl
 
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 7061d2f0af1b7f475566d65b68f59c655fe22507..bea7a2fc6aa76bf2aa2510e4adc2ca31654312ce 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2563,4 +2563,12 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         return this.spigot;
     }
     // Spigot end
+
+    // CloudPlane start - expose tracked entities
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T extends org.bukkit.entity.Entity> Collection<? extends T> getTrackedEntities(Class<T> entityClass) {
+        return entityClass == Player.class ? (Set<T>) getTrackedPlayers() : getWorld().getNearbyEntitiesByType(entityClass, getLocation(), entity.tracker.range);
+    }
+    // CloudPlane end
 }
