From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SuperKooks <superkooks@superkooks.com>
Date: Wed, 4 Aug 2021 18:05:01 +1000
Subject: [PATCH] Expose tracked entities


diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
index ff52a502c2e0f93d42aa20f2c34cc5414c97b11a..e55faf4bea1cf89e2b4eeb3990c80efdc122beb3 100644
--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
+++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
@@ -2211,7 +2211,7 @@ public class ChunkMap extends ChunkStorage implements ChunkHolder.PlayerProvider
 
         final ServerEntity serverEntity;
         final Entity entity;
-        private final int range;
+        public final int range; // CloudPlane - private --> public
         SectionPos lastSectionPos;
         public final Set<ServerPlayerConnection> seenBy = new ReferenceOpenHashSet<>(); // Paper - optimise map impl
 
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 557f594cf619379dac1389ca483d7046a7c689e2..e8716200a7a28b9ba218e9c5db8657c7c1ca8c4a 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2609,4 +2609,12 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         return this.spigot;
     }
     // Spigot end
+
+    // CloudPlane start - expose tracked entities
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T extends org.bukkit.entity.Entity> Collection<? extends T> getTrackedEntities(Class<T> entityClass) {
+        return entityClass == Player.class ? (Set<T>) getTrackedPlayers() : getWorld().getNearbyEntitiesByType(entityClass, getLocation(), entity.tracker.range);
+    }
+    // CloudPlane end
 }
