From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SuperKooks <superkooks@superkooks.com>
Date: Wed, 4 Aug 2021 18:05:01 +1000
Subject: [PATCH] Expose tracked entities


diff --git a/src/main/java/net/minecraft/server/level/ChunkMap.java b/src/main/java/net/minecraft/server/level/ChunkMap.java
index 049ed12d1e74f8ac5937e06047806acdc7967d5f..4c55cd3cd47ea2452b95c0eff6b620ece44f5709 100644
--- a/src/main/java/net/minecraft/server/level/ChunkMap.java
+++ b/src/main/java/net/minecraft/server/level/ChunkMap.java
@@ -2285,7 +2285,7 @@ Sections go from 0..16. Now whenever a section is not empty, it can potentially
 
         public final ServerEntity serverEntity; // Purpur -> package -> public
         final Entity entity;
-        private final int range;
+        public final int range; // CloudPlane - private --> public
         SectionPos lastSectionPos;
         public final Set<ServerPlayerConnection> seenBy = new ReferenceOpenHashSet<>(); // Paper - optimise map impl
 
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index bc00a03aff2f644f15c56cc3206715437db6bc07..897f270b53f1382dff9cbaa39ac3e25c64a6624d 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2689,4 +2689,12 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         return getHandle().flyingFallDamage;
     }
     // Purpur end
+
+    // CloudPlane start - expose tracked entities
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T extends org.bukkit.entity.Entity> Collection<? extends T> getTrackedEntities(Class<T> entityClass) {
+        return entityClass == Player.class ? (Set<T>) getTrackedPlayers() : getWorld().getNearbyEntitiesByType(entityClass, getLocation(), entity.tracker.range);
+    }
+    // CloudPlane end
 }
