From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jake Potrebic <jake.m.potrebic@gmail.com>
Date: Thu, 22 Jul 2021 21:43:20 -0700
Subject: [PATCH] Add back link detection on non-chat event messages


diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 710d54e777ad8ce69e979eed3e1b3174c65dfcd0..c43af6db5947b771f971e5c514d38bb0ab84f52c 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -1469,6 +1469,7 @@ public abstract class PlayerList {
     public void broadcastMessage(Component message, ChatType type, UUID sender) {
         this.server.sendMessage(message, sender);
         Iterator iterator = this.players.iterator();
+        if (tk.booky.cloudplane.CloudPlaneConfig.linkDetectionNonAdventure) message = CraftChatMessage.fixComponent(message); // CloudPlane - add back URL detection
 
         while (iterator.hasNext()) {
             ServerPlayer entityplayer = (ServerPlayer) iterator.next();
diff --git a/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java b/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
index 33a1c39023806b5fb8eb96be2f9ffa42367a2d37..cac3c3e0776ce50641d7ccfa33273e2c3afab820 100644
--- a/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
+++ b/src/main/java/tk/booky/cloudplane/CloudPlaneConfig.java
@@ -165,6 +165,7 @@ public class CloudPlaneConfig {
     }
 
     public static boolean localizeItems = false;
+    public static boolean linkDetectionNonAdventure = false;
     private static void adventure() {
         if (version == 2) {
             localizeItems = getBoolean("settings.localize.items", localizeItems);
@@ -172,6 +173,7 @@ public class CloudPlaneConfig {
             set("settings.localize", null);
         } else {
             localizeItems = getBoolean("settings.adventure.localize-items", localizeItems);
+            linkDetectionNonAdventure = getBoolean("settings.adventure.link-detection-non-adventure", linkDetectionNonAdventure);
         }
     }
 
